java/src/processing/mode/java/runner/Runner.java

```diff
@@ -90,9 +90,19 @@ public Runner(JavaBuild build, RunnerListener listener) throws SketchException {
 
     // Make sure all the imported libraries will actually run with this setup.
     int bits = Base.getNativeBits();
+    String arch = Base.getNativeArch();
```

Comment:
Rather than doing this a second time here, let's do methods in Base:
```
static public String getVariant(String arch, int bits) {
  // all the logic goes here
}

static public String getVariant() {
  return getVariant(getNativeBits(), getNativeArch());
}
```